File: VB/13_Conversion/EmbedSVG.vb
Imports Spire.Pdf

Namespace EmbedSVG
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the input file path
			Dim file As String = "..\..\..\..\..\..\..\Data\SampleB_1.pdf"

			' Open the PDF document
			Dim doc As New PdfDocument()
			doc.LoadFromFile(file)

			' Set the "useEmbeddedSvg" boolean flag to true to convert to HTML with embedding SVG
			doc.ConvertOptions.SetPdfToHtmlOptions(True)

			' Specify the output file name for the converted HTML
			Dim result As String = "ToHTMLWithEmbeddingSVG_out.html"

			' Convert the document to an HTML file
			doc.SaveToFile(result, FileFormat.HTML)

			' Close the PDF document
			doc.Close()

			' Launch result file
			PDFDocumentViewer(result)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace

File: CS/13_Conversion/EmbedSVG.cs
using Spire.Pdf;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace EmbedSVG
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Specify the path of the input PDF file
            String file = @"..\..\..\..\..\..\..\Data\SampleB_1.pdf";

            // Open the PDF document using PdfDocument
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile(file);

            // Set the boolean option to embed SVG when converting to HTML
            doc.ConvertOptions.SetPdfToHtmlOptions(true);

            // Specify the result file name for the converted HTML file
            String result = "ToHTMLWithEmbedingSVG_out.html";

            // Convert the PDF document to an HTML file
            doc.SaveToFile(result, FileFormat.HTML);

            // Close the PDF document
            doc.Close();

            //Launch result file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }   

    }
}
